### DISCUSSION TITLE
Make "Docs" and "Blog" their own files within /src/pages just like any other Page (vs. hiding docs & blogs pages src files)

### DISCUSSION THREAD

**Discussion by pmeaney** [facebook/docusaurus#9735] [General]
I see that creating a new page is quite easy (https://docusaurus.io/docs/creating-pages).
However-- I want to edit the existing, default pages of "Docs" and "Blog". 

 This way I can change the structure or styling.  

For example, to use TailwindCSS with Docusaurus, I need to add the "prose" class to the html div which contains the div containing markdown.  Otherwise, the markdown is not rendered properly due to tailwindcss overriding its styles (https://stackoverflow.com/a/75709690 , https://stackoverflow.com/a/68927527).

I think it's very nice that Docusaurus makes it easy to add a new page.  

**However Docusaurus should follow that convention by default! There should be a Blog.js and a Docs.js within /src/pages/ ! These should not be hidden! I want to be able to edit the style of all of my Docusaurus content, so please-- do not design this system in a way that parts are hidden from the developer and cannot have their style edited.** 

For example, I don't want to use https://infima.dev/ .  I should be able to edit every page so that I can change all html tags' classes to my liking.  Please don't hide things from users (such as Docs or Blogs page .js files) or force things on users (such as by hiding files and only making their build files accessible but not src files, forcing us to use your style choices).  Make everything editable and customizable.

**Comment by slorber** [comment]
Everything is customizable thank to the swizzle feature. Hiding the files is done on purpose.

The code required to render blog posts and docs is not a single 100 lines of code file.

See our theme source code for example, it's a lot of code, and only a subset of it: 
https://github.com/facebook/docusaurus/tree/main/packages/docusaurus-theme-classic/src/theme

Having all these files by default in your docusaurus project would be overwhelming, and would end up becoming stale over time as we improve Docusaurus. 

We are a docs tool. We are not a docs boilerplate.

Note that we have a plugin system, so if you don't like our default approach to docs you can also create your own opinionated system.

**Reply by pmeaney** [reply]
Thanks! I appreciate the info.

Given that all of Docs & Blogs can be hidden...

Would it be possible to hide **most** of Docs & Blogs, yet leave available for editing their Layout file?  Or in some way, allow Developers to edit their html div class names to add/remove styling?

or, if we must use Swizzle to edit those files (e.g. layout)-- Would it be possible to, in the future, Swizzle in a "(safe)" way (rather than "unsafe") the Docs and Blog layout files?

Thanks!
